app/routers/google_connect.py

```diff
@@ -0,0 +1,261 @@
+
+    service = build('calendar', 'v3', credentials=credentials)
+    events_result = service.events().list(
+        calendarId='primary',
+        timeMin=start,
+        timeMax=end,
+        singleEvents=True,
+        orderBy='startTime'
+    ).execute()
+
+    events = events_result.get('items', [])
+    return events
+
+
+def push_events_to_db(events: list, user: User,
+                      session: SessionLocal = Depends()) -> bool:
+    '''Adding google events to db'''
+    db_cleanup(user, session)
+
+    for event in events:
+        # Running over the events that have come from the API
+        location = None
+        title = event['summary']  # The Google event title
+
+        # support for all day events
+        if 'dateTime' in event['start'].keys():
+            # This case handles part time events (not all day events)
+            start = datetime.fromisoformat(event['start']['dateTime'])
+            end = datetime.fromisoformat(event['end']['dateTime'])
+        else:
+            # This case handles all day events
+            # It takes the date string in this format yyyy-mm-dd
+            # - according to the documentation
+            # and then split it and passing it to datetime constructor.
+            start_in_str = event['start']['date'].split('-')
+            start = datetime(
+                year=int(start_in_str[0]),
+                month=int(start_in_str[1]),
+                day=int(start_in_str[2])
+            )
+
+            end_in_str = event['end']['date'].split('-')
+            end = datetime(
+                year=int(end_in_str[0]),
+                month=int(end_in_str[1]),
+                day=int(end_in_str[2])
+            )
+
+        if 'location' in event.keys():
+            # This case handles if Google Event has a location attached
+            location = event['location']
+
+        create_google_event(title, start, end, user, location, session)
+    return True
+
+
+def create_google_event(title: str, start: datetime,
+                        end: datetime, user: User, location: str,
+                        session: SessionLocal = Depends()) -> Event:
+    return create_event(
+        # creating an event obj and pushing it into the db
+        db=session,
+        title=title,
+        start=start,
+        end=end,
+        owner_id=user.id,
+        location=location,
+        is_google_event=True
+    )
+
+
+def db_cleanup(user: User, session: SessionLocal = Depends()) -> bool:
+    '''removing all user google events so the next time will be syncronized'''
+
+    for user_event in user.events:
+        user_event_id = user_event.id
+        event = user_event.events
+        if event.is_google_event:
+            session.query(Event).filter_by(id=event.id).delete()
+            session.query(UserEvent).filter_by(id=user_event_id).delete()
+            session.commit()
+
+    return True
+
+
+def get_credentials_from_db(user: User) -> tuple:
```

Comment:
Change to `Tuple`, and specify the types of the elements

Reply:
Done.